vsphere-mixin/variables.libsonnet

```diff
@@ -0,0 +1,236 @@
+      local vmRegex = '/vcenter_vm_name="(?<value>[^"]*)",vm_path="(?<text>[^"]*)"/',
+
+      local varMetric = 'vcenter_vm_memory_usage_mebibytes',
+      local topResourceSelector =
+        var.custom.new(
+          'top_resource_count',
+          values=[2, 4, 6, 8, 10],
+        )
+        + var.custom.generalOptions.withDescription(
+          'This variable allows for modification of top resource value.'
+        )
+        + var.custom.generalOptions.withLabel('Top resource count'),
+      local root = self,
+      local groupVariablesFromLabels(groupLabels, filteringSelector) =
+        local chainVarProto(index, chainVar) =
+          var.query.new(chainVar.label)
+          + var.query.withDatasourceFromVariable(root.datasources.prometheus)
+          + var.query.queryTypes.withLabelValues(
+            chainVar.label,
+            '%s{%s}' % [varMetric, chainVar.chainSelector],
+          )
+          + var.query.generalOptions.withLabel(extendedUtils.toSentenceCase(chainVar.label))
+          + var.query.selectionOptions.withIncludeAll(
+            value=true,
+            customAllValue='.+'
+          )
+          + var.query.selectionOptions.withMulti(
+            true,
+          )
+          + var.query.refresh.onTime()
+          + var.query.withSort(
+            i=1,
+            type='alphabetical',
+            asc=true,
+            caseInsensitive=false
+          );
+        std.mapWithIndex(chainVarProto, utils.chainLabels(groupLabels, [filteringSelector])),
+      datasources: {
+        prometheus:
+          var.datasource.new('prometheus_datasource', 'prometheus')
+          + var.datasource.generalOptions.withLabel('Data source')
+          + var.datasource.withRegex(''),
+        loki:
+          var.datasource.new('loki_datasource', 'loki')
+          + var.datasource.generalOptions.withLabel('Loki data source')
+          + var.datasource.withRegex('(?!grafanacloud.+usage-insights|grafanacloud.+alert-state-history).+')
```

Comment:
I think this is added automatically on the integration level.

Reply:
Just a note here, it is, but it then breaks when using Grizzly, so something we all need to be aware of when reviewing dashboards :)